staging/src/k8s.io/kubectl/pkg/cmd/version/version.go

```diff
@@ -120,7 +121,7 @@ func (o *Options) Run() error {
 	clientVersion := version.Get()
 	versionInfo.ClientVersion = &clientVersion
 
-	if !o.ClientOnly && o.discoveryClient != nil {
+	if !o.ClientOnly && !reflect.ValueOf(o.discoveryClient).IsNil() {
```

Comment:
I doubt this is the right fix, the usage of reflection here is unnecessary. The code is identical to the previous check, imo.

Reply:
It isn't :/ I've faced the error while using GCP Authentication. 

Here's an explanation: https://glucn.medium.com/golang-an-interface-holding-a-nil-value-is-not-nil-bb151f472cc7

The interface is not nil, but the value of discoveryClient is initialized (so it's not nil) but with nil value.

Anyway I can try to figure out other solution for this

Reply:
Based on that reading it looks like you're stumbling on a half baked `o.discoveryClient`. Not sure how you set up, but in this command this could happen if this invocation https://github.com/kubernetes/kubernetes/blob/a8124860d2211943ba955d5953c01452b4bacfee/staging/src/k8s.io/kubectl/pkg/cmd/version/version.go#L94 would return nil error, but half baked discovery client, ie. one that is satisfying interface but is actually nil. I'd suggest checking your invocation tree, it's possible that something is off earlier in the code.